### ISSUE TITLE
Some problems on depth to point cloud

### ISSUE THREAD

**Issue by Jiahao620** [facebookresearch/sapiens#81]
Thanks for your nice work!
I face some problems when I try to convert the predicted depthmap to the point cloud.
The depthmap visualization seems good, but the point cloud from depthmap is weird. Here is my visualization of point cloud and my unprojection  code. I implement it with a pseudo camera.
```
def get_pts3d(depthmap, w2c, intrins):
    depthmap = torch.tensor(depthmap).cuda()
    mask = torch.tensor(~ np.isnan(depthmap.cpu().numpy()))
    c2w = np.linalg.inv(w2c)
    c2w = torch.tensor(c2w).cuda()

    W, H = depthmap.shape[1], depthmap.shape[0]
   
    intrins = torch.tensor(K).float().cuda()
    grid_x, grid_y = torch.meshgrid(torch.arange(W), torch.arange(H), indexing='xy')
    points = torch.stack([grid_x, grid_y, torch.ones_like(grid_x)], dim=-1).reshape(-1, 3).float().cuda()
    rays_d = points @ intrins.inverse().T @ c2w[:3 , :3].T
    rays_o = c2w[:3, 3]

    pts_all = depthmap.reshape(-1, 1)* rays_d * 10 + rays_o

    pts3d = pts_all[mask.reshape(-1)]
    return pts_all, pts3d
```
![depthmap](https://github.com/user-attachments/assets/257bf537-dbd0-40cd-98bb-e6a4eca527ef)
![image](https://github.com/user-attachments/assets/5e9cf128-cf55-4a48-85bf-b8125f4a84b7)
I'm looking forward to your instructions!

**Comment by rawalkhirodkar** [comment]
@Jiahao620 the predicted depth is relative.
You would need scale and translation (two scalars) to transform the relative depth to absolute depth.
abs_depth = rel_depth * s + t.

In this example: https://rawalkhirodkar.github.io/sapiens/static/videos/iphone_vs_sapiens.mp4
We obtain the scale and translation by aligning the predictions with another metric reference (from the phone lidar). 
Without this, unfortunately, you would have to guess the scale and translation to obtain a good point cloud.
